commons-macros/src/main/scala/com/avsystem/commons/macros/serialization/GenCodecMacros.scala

```diff
@@ -258,30 +265,66 @@ class GenCodecMacros(ctx: blackbox.Context) extends CodecMacroCommons(ctx) with
       def generatedWrite(sym: Symbol) =
         q"writeField(${nameBySym(sym)}, output, ${mkParamLessCall(q"value", sym)}, ${depNames(sym)})"
 
-      q"""
-        new $baseClass with $GenCodecObj.ErrorReportingCodec[$tpe] {
-          ..${params.filterNot(ap => oooFields.contains(nameBySym(ap.sym))).map(depDeclaration)}
-          ..${generated.collect({ case (sym, depTpe) if !oooFields.contains(nameBySym(sym)) => generatedDepDeclaration(sym, depTpe) })}
-          $emptyOooFieldsDecl
-          protected def typeRepr = ${tpe.toString}
-          def nullable = ${typeOf[Null] <:< tpe}
-          def readObject(input: $SerializationPkg.ObjectInput, ..$oooFieldsParam): $tpe = {
-            ..${params.map(varDeclaration)}
-            while(input.hasNext) {
-              val fi = input.nextField()
-              fi.fieldName match {
-                case ..${params.map(fieldReadCase)}
-                case _ => fi.skip()
+      if (!isScalaJs) {
```

Comment:
Please reduce code duplication and extract the parts that differ between JS and JVM in more finegrained manner.